gdc_client/upload/parser.py

```diff
@@ -9,15 +9,20 @@
 from . import exceptions
 
 from .client import GDCUploadClient
+from .. import log
 
 
+logger = log.get_logger('upload-client')
+
 def validate_args(parser, args):
     """ Validate argparse namespace.
     """
-    if args.manifest or args.identifier:
+    if args.manifest or args.file_ids:
+        if args.identifier:
```

Comment:
that was the old help message. This is no longer what shows when you print the help message. The `identifier` is kept only for backwards compatibility and is completely useless and ignored (except to print a warning).
I didn't print the warning if both `manifest` and `file_ids` were missing because I figured that the user has bigger problems to deal with if that's the case. I mean, I don't see a user specifying `-i` or `--identifier` without also specifying a `file_id` and if they do well the real problem is that they don't specify the `file_id`. The use of the `identifier` is deprecated but not an error per say.

In any case, I changed it to warn the user before exiting so they can fix that as well as fixing the missing `manifest` or `file_id`.